mailpoet/lib/Util/Notices/SenderDomainAuthenticationNotices.php

```diff
@@ -0,0 +1,167 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Util\Notices;
+
+use MailPoet\Services\AuthorizedSenderDomainController;
+use MailPoet\Services\Bridge;
+use MailPoet\Settings\SettingsController;
+use MailPoet\Util\FreeDomains;
+use MailPoet\Util\Helpers;
+use MailPoet\Util\License\Features\Subscribers;
+use MailPoet\WP\Notice;
+
+class SenderDomainAuthenticationNotices {
+
+  const LOWER_LIMIT = 500;
+  const UPPER_LIMIT = 1000;
+
+  const FREE_MAIL_KB_URL = 'https://kb.mailpoet.com/article/259-your-from-address-cannot-be-yahoo-com-gmail-com-outlook-com';
+  const SPF_DKIM_DMARC_KB_URL = 'https://kb.mailpoet.com/article/295-spf-dkim-dmarc';
+
+  private SettingsController $settingsController;
+
+  private Subscribers $subscribersFeatures;
+
+  private FreeDomains $freeDomains;
+
+  private AuthorizedSenderDomainController $authorizedSenderDomainController;
+
+  private Bridge $bridge;
+
+  public function __construct(
+    SettingsController $settingsController,
+    Subscribers $subscribersFeatures,
+    FreeDomains $freeDomains,
+    AuthorizedSenderDomainController $authorizedEmailsController,
+    Bridge $bridge
+  ) {
+    $this->settingsController = $settingsController;
+    $this->subscribersFeatures = $subscribersFeatures;
+    $this->freeDomains = $freeDomains;
+    $this->authorizedSenderDomainController = $authorizedEmailsController;
+    $this->bridge = $bridge;
+  }
+
+  public function getDefaultFromAddress(): string {
+    return $this->settingsController->get('sender.address', '');
+  }
+
+  public function getDefaultFromDomain(): string {
+    return Helpers::extractEmailDomain($this->getDefaultFromAddress());
+  }
+
+  public function isFreeMailUser(): bool {
+    return $this->freeDomains->isEmailOnFreeDomain($this->getDefaultFromDomain());
+  }
+
+  public function init($shouldDisplay): ?Notice {
+    if (
+      !$shouldDisplay
+      || !$this->bridge->isMailpoetSendingServiceEnabled()
```

Comment:
The banners should not be displayed for new users.
I might have missed it but I don't seem to find where the condition is implemented?

Reply:
How are we defining new users?  `$shouldDisplay` is defined here, similar to many other notices: https://github.com/mailpoet/mailpoet/blob/MAILPOET-5783%2Fsender-domain-banner-notices/mailpoet/lib/Util/Notices/PermanentNotices.php#L147-L147

It should prevent the notice from being displayed during the welcome wizard. Is there a different check we should include?

Reply:
> How are we defining new users?

The idea is to show the banners only to users that were already sending emails before the new requirements were applied.

New users will see the onboarding task and the inline notices when the start creating emails but existing users might not go inside the editor or to the settings page, so we need to make them aware of the changes.

New users are users that have not sent an email yet.

Reply:
@brezocordero 3db27727fe21715d323c8e1fb08d6224206f1814 should ensure that we don't display these notices for new users.